llvm/lib/Target/RISCV/RISCVInstrInfo.cpp

```diff
@@ -4191,6 +4191,24 @@ bool RISCVInstrInfo::simplifyInstruction(MachineInstr &MI) const {
       return true;
     }
     break;
+  case RISCV::BLTU:
+    // bltu zero, rs, imm => bne rs, zero, imm
+    if (MI.getOperand(0).getReg() == RISCV::X0) {
+      MachineOperand MO0 = MI.getOperand(0);
+      MI.removeOperand(0);
+      MI.insert(MI.operands_begin() + 1, {MO0});
+      MI.setDesc(get(RISCV::BNE));
+    }
+    break;
+  case RISCV::BGEU:
+    // bgeu zero, rs, imm => beq rs, zero, imm
```

Comment:
Thinking about other cases here:

bgeu rs, zero, imm is an unconditionally taken branch.  Rewriting as a unconditional branch is hard (due to CFG updates), but maybe we could turn this into beq zero, zero, imm?  That would remove the register use, and might compress better?

Reply:
`beq zero, zero, imm` isn't compressible. The first register needs to be x8-x15.

Reply:
Yes, `bgeu reg, zero` is on my radar and as you say the CFG update makes it challenging. I hadn't considered canonicalising to `beq zero, zero, imm`. `beq reg, reg, imm` would be an alternative that has the chance of being compressible.

There are ~440 static instances of `bgeu reg, zero` in an RVA22 test-suite build. 40 instances of `beq` with rs1 == rs2 and 50 of `bne` with rs1 == rs2.